packages-web/maps/src/components/MapsContainer.ts

```diff
@@ -220,16 +220,18 @@ class MapsContainer extends Component<MapsContainerProps, MapsContainerState> {
         );
     }
 
-    private onClickMarker = (event: LeafletEvent & google.maps.MouseEvent, locationAttr: DataSourceLocationProps) => {
+    private onClickMarker = (event: LeafletEvent & GoogleMapsClickEvent, locationAttr: DataSourceLocationProps) => {
         const { locations } = this.state;
-        const latitude = this.props.mapProvider === "googleMaps" ? event.latLng.lat() : event.target.getLatLng().lat;
-        const longitude = this.props.mapProvider === "googleMaps" ? event.latLng.lng() : event.target.getLatLng().lng;
-        this.executeAction(
-            locations[
-                locations.findIndex(targetLoc => targetLoc.latitude === latitude && targetLoc.longitude === longitude)
-            ],
-            locationAttr
-        );
+        const GUID = this.props.mapProvider === "googleMaps" ? event.options.GUID : event.target.options.GUID;
```

Comment:
Can we make this symmetric for all providers?

Reply:
Nope because the way we are receiving the props are different...